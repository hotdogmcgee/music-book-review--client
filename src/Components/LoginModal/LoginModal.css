.modal {
  position: fixed;
  top: 0;
  left: 0;
  display: block;
  width: 100%;
  height: 100%;
  background: rgba(0, 0, 0, 0.3);
}

.modal-main {
  position: fixed;
  background: white;
  width: 80%;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}

.display-block {
  display: block;
}

.display-none {
  display: none;
}

.login-modal {
  height: 462px;
  overflow-y: scroll;
}

.login-modal input {
  width: 100%;
}

.login-modal ul {
  list-style: none;
  display: flex;
  justify-content: space-around;
  padding: 0;
  border-bottom: 1px solid black;
  margin-top: 0;
  margin-bottom: 2em;
  text-align: center;
}

.modal-tab:first-child {
  border-right: 1px solid black;
}

.modal-tab {
  width: 50%;
  cursor: pointer;
}

.highlight {
  background-color: #d26e52;
}

.SuccessIcon__container {
  display: flex;
  align-items: center;
  height: inherit;
  flex-direction: column;
  justify-content: center;
  background-color: #373a47;
}

.SuccessIcon__container > h2 {
  color: white;
  text-align: center;
}

@media screen and (min-width:900px) {
  .modal-main {
    width: 600px;
    overflow-y: hidden;
    height: auto
  }

  .SuccessIcon__container {
    height: 462px
  }


}
