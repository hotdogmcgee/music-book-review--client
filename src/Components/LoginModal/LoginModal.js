import React from "react";
import "./LoginModal.css";
import LoginForm from "../LoginForm/LoginForm";
import RegisterForm from "../RegisterForm/RegisterForm";
import SuccessIcon from "../SuccessIcon/SuccessIcon";

export default class LoginModal extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      formType: "log-in",
      loginSuccess: false,
      successMessage: ""
    };
  }

  setForm = type => {
    const formType = type;
    this.setState({ formType: formType });
  };

  setLoginSuccess = () => {
    this.setState({ loginSuccess: !this.state.loginSuccess });
  };

  setSuccessMessage = message => {
    this.setState({ successMessage: message });
  };

  handleLoginSuccess = () => {
    const showAfterSuccess = () => {
      this.props.handleClose();
      this.setLoginSuccess();
    };

    this.setSuccessMessage("Success!");

    this.props.hasLogin(true);
    this.setLoginSuccess();
    setTimeout(() => showAfterSuccess(), 3000);
  };

  handleRegisterSuccess = () => {
    const showAfterSuccess = () => {
      this.props.handleClose();
      this.setLoginSuccess();
    };

    this.setSuccessMessage("Registration complete, you are logged in as well!");

    this.props.hasLogin(true);
    this.setLoginSuccess();
    setTimeout(() => showAfterSuccess(), 3000);
  };

  renderSuccessIcon() {
    return (
      <div className="SuccessIcon__container">
        <SuccessIcon />
        <h2>{this.state.successMessage}</h2>
      </div>
    );
  }

  renderLoginModal() {
    const loginClasses =
      this.state.formType === "log-in" ? " modal-tab highlight" : "modal-tab";
    const registerClasses =
      this.state.formType === "register" ? " modal-tab highlight" : "modal-tab";
    const { handleClose } = this.props;
    return (
      <div>
        <ul className="cd-signin-modal__switcher js-signin-modal-switcher js-signin-modal-trigger">
          <li onClick={() => this.setForm("log-in")} className={loginClasses}>
            <p>Log In</p>
          </li>
          <li
            onClick={() => this.setForm("register")}
            className={registerClasses}
          >
            <p>Register</p>
          </li>
        </ul>
        {this.state.formType === "log-in" ? (
          <LoginForm
            onLoginSuccess={this.handleLoginSuccess}
            handleClose={handleClose}
          />
        ) : (
          <RegisterForm
            handleClose={handleClose}
            onRegisterSuccess={this.handleRegisterSuccess}
          />
        )}
      </div>
    );
  }

  render() {
    const { show } = this.props;
    const showHideClassName = show
      ? "modal display-block"
      : "modal display-none";
    return (
      <div className={showHideClassName}>
        <section className="modal-main login-modal">
          {this.state.loginSuccess
            ? this.renderSuccessIcon()
            : this.renderLoginModal()}
        </section>
      </div>
    );
  }
}
