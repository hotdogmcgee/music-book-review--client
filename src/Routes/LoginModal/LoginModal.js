import React from "react";
import ReactDOM from "react-dom";

import "./LoginModal.css";

// const LoginModal = (props) => {
//     return (
//         <div>
//             <div className="modal-wrapper"
//                 style={{
//                     transform: props.show ? 'translateY(0vh)' : 'translateY(-100vh)',
//                     opacity: props.show ? '1' : '0'
//                 }}>
//                 <div className="modal-header">
//                     <h3>Modal Header</h3>
//                     <span className="close-modal-btn" onClick={props.close}>×</span>
//                 </div>
//                 <div className="modal-body">
//                     <p>
//                         {props.children}
//                     </p>
//                 </div>
//                 <div className="modal-footer">
//                     <button className="btn-cancel" onClick={props.close}>CLOSE</button>
//                     <button className="btn-continue">CONTINUE</button>
//                 </div>
//             </div>
//         </div>
//     )
// }

// export default LoginModal;

// const LoginModal = ({ children, onClose, open }) =>
//   open
//   ? ReactDOM.createPortal(
//     <div className="modal">
//       <div className="modal__close" onClick={onClose}>&times;</div>
//       {children}
//     </div>,
//     document.body
//   )
//   :null

//   export default LoginModal

// function makeTabs() {
//   return (
//     <ul id="myTab" className="nav nav-tabs">
//               <li id="tab1" className=" active tab-style login-shadow "><a href="#signin" data-toggle="tab">Log In</a></li>
//               <li id="tab2" className=" tab-style "><a href="#signup" data-toggle="tab">Sign Up</a></li>

//             </ul>
//   )
// }

const LoginModal = ({ children, onClose, open }) =>
  open
    ? ReactDOM.createPortal(
        <div>
          <div
            className="modal-wrapper"
            style={{
              transform: open ? "translateY(0vh)" : "translateY(-100vh)",
              opacity: open ? "1" : "0"
            }}
          >
            {/* <div className="modal-header">
              <h3>Modal Header</h3>
              <span className="close-modal-btn" onClick={onClose}>
                ×
              </span>
            </div> */}
            <div className="modal-body">
              <div className="RML-login-modal-box">
                <div className="RML-login-modal-box-content">
                  <div className="RML-login-modal-close">X</div>
                  <div className="RML-login-modal-mode">
                    <div className=" active">Sign in</div>
                    <div className="">Sign up</div>
                  </div>
                  <div className="RML-social-modal-content-wrap">
                    <div className="RML-login-modal-form">
                      <div className="RML-form-group">
                        <label htmlFor="form-email">Email</label>
                        <input
                          type="email"
                          className="RML-form-control"
                          id="form-email"
                          name="email"
                          placeholder="Email"
                          defaultValue=""
                        ></input>
                      </div>
                      <div className="RML-form-group">
                        <label htmlFor="form-password">Password</label>
                        <input
                          type="password"
                          className="RML-form-control"
                          id="form-password"
                          name="password"
                          placeholder="Password"
                          defaultValue=""
                        ></input>
                      </div>
                      <button className="RML-btn" id="loginSubmit">
                        Sign in
                      </button>
                      {/* <div class="clearfix"></div> */}
                    </div>
                  </div>
                </div>
              </div>
            </div>
            {/* <div className="modal-footer">
              <button className="btn-cancel" onClick={onClose}>
                CLOSE
              </button>
              <button className="btn-continue">CONTINUE</button>
            </div> */}
          </div>
        </div>,
        document.body
      )
    : null;

export default LoginModal;
