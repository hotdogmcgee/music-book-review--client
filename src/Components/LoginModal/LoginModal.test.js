import React from "react";
import LoginModal from "./LoginModal";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("LoginModal component", () => {
  it("renders a LoginModal by default", () => {
    const wrapper = shallow(<LoginModal />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
