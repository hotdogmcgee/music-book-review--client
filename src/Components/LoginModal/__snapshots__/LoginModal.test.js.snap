// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginModal component renders a LoginModal by default 1`] = `
<div
  className="modal display-none"
>
  <section
    className="modal-main login-modal"
  >
    <div>
      <ul
        className="cd-signin-modal__switcher js-signin-modal-switcher js-signin-modal-trigger"
      >
        <li
          className=" modal-tab highlight"
          onClick={[Function]}
        >
          <p>
            Log In
          </p>
        </li>
        <li
          className="modal-tab"
          onClick={[Function]}
        >
          <p>
            Register
          </p>
        </li>
      </ul>
      <LoginForm
        onLoginSuccess={[Function]}
      />
    </div>
  </section>
</div>
`;
